File: 2021/1/part2.py
def part2(filename):
	with open(filename, 'r') as file:
		data = [int(l) for l in file.read().splitlines()]

	count = 0

	for i in range(1, len(data) - 2):
		prev = sum(data[i-1:i+2])
		curr = sum(data[i:i+3])

		if curr > prev:
			count += 1

	return count

if __name__ == '__main__':
	count = part2('input.txt')
	print(count)

File: 2024/01/part2.go
package main

func part2(l []int, r []int) int {
	result := 0
	for _, lval := range l {
		count := Count(r, lval)
		result += lval * count
	}
	return result
}

func Count(arr []int, target int) int {
	count := 0
	for _, v := range arr {
		if v == target {
			count++
		}
	}
	return count
}
